providers/github/provider/provider.go

```diff
@@ -46,17 +46,35 @@ func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error)
 		Discover: &inventory.Discovery{},
 	}
 
+	if x, ok := flags["enterprise-url"]; ok && len(x.Value) != 0 {
+		conf.Options[connection.OPTION_ENTERPRISE_URL] = string(x.Value)
+	}
+
+	isAppAuth := false
+	if appId, ok := req.Flags[connection.OPTION_APP_ID]; ok && len(appId.Value) > 0 {
+		conf.Options[connection.OPTION_APP_ID] = string(appId.Value)
+
+		installId := req.Flags[connection.OPTION_APP_INSTALLATION_ID]
+		conf.Options[connection.OPTION_APP_INSTALLATION_ID] = string(installId.Value)
+
+		pk := req.Flags[connection.OPTION_APP_PRIVATE_KEY]
+		conf.Options[connection.OPTION_APP_PRIVATE_KEY] = string(pk.Value)
+		isAppAuth = true
```

Comment:
If the app-id is set but installation id or private key are missing we should return an error

Reply:
that's done in the connection, not here. I wanted to not duplicate the logic for checking if everything is set in the `ParseCLI`

Reply:
That makes a lot of sense.